src/FieldWarning/Assets/Units/Component/Movement/PathfinderData.cs

```diff
@@ -56,15 +57,16 @@ public class PathfinderData : Loader
         ///     A sampling of the terrain topology, 
         ///     used if generating from scratch.
         /// </param>
-        public PathfinderData(TerrainMap map)
+        public PathfinderData(TerrainMap map, List<MobilityType> mobilityTypes)
```

Comment:
I think its fair to pass in the entire armory here.. the armory should be able to give us the mobility types or we can filter it from the armory

Reply:
The idea is to pass the minimum necessary, especially since it's stored as a field of the object. If PathfinderData has a field with the mobility types it's kinda obvious why there might be a connection/dependency. If it just has a handle to the entire armory then you'd have to wonder or dig a bit more to find what the variable is used for.